app/controllers/admin/volunteers.py

```diff
@@ -92,10 +99,22 @@ def addVolunteer(eventId):
         user = User.get(User.username==user)
         isVolunteerInEvent =  (EventRsvp.select().where(EventRsvp.user==user, EventRsvp.event_id == eventId).exists() and
               EventParticipant.select().where(EventParticipant.user == user, EventParticipant.event_id == eventId).exists())
-
+        userAdded = EventRsvp.get_or_none(user = user, event = eventId)
         if len(eventParticipants) == 0 or isVolunteerInEvent == False:
-            addVolunteerToEventRsvp(user, eventId)
-            EventParticipant.create(user = user, event = eventId)
+            if event.isPast:
+                eventHours = getEventLengthInHours(event.timeStart, event.timeEnd, event.startDate)
+                if not (userAdded):
+                    EventParticipant.create(user = user, event = eventId, hoursEarned = eventHours)
```

Comment:
We are we only creating this record if they haven't rsvp'd? This is confusing to me.